src/System.Security.Cryptography.OpenSsl/src/System/Security/Cryptography/RSAOpenSsl.cs

```diff
@@ -45,6 +45,31 @@ public RSAOpenSsl(RSAParameters parameters)
             ImportParameters(parameters);
         }
 
+        /// <summary>
+        /// Create an RSAOpenSsl from an existing <see cref="IntPtr"/> whose value is an
+        /// existing OpenSSL <c>RSA*</c>.
+        /// </summary>
+        /// <remarks>
+        /// This method will increase the reference count of the <c>RSA*</c>, the caller should
+        /// continue to manage the lifetime of their reference.
+        /// </remarks>
+        /// <param name="handle">A pointer to an OpenSSL <c>RSA*</c></param>
+        /// <exception cref="ArgumentException"><paramref name="handle" /> is invalid</exception>
+        public RSAOpenSsl(IntPtr handle)
+        {
+            if (handle == IntPtr.Zero)
+                throw new ArgumentException(SR.Cryptography_OpenInvalidHandle, "handle");
+
+            _legalKeySizesValue = new[] { s_legalKeySizes };
+
+            SafeRsaHandle rsaHandle = SafeRsaHandle.DuplicateHandle(handle);
+
+            // Set base.KeySize to avoid throwing an extra Lazy at the GC when
+            // using something other than the default keysize.
+            base.KeySize = 8 * Interop.libcrypto.RSA_size(rsaHandle);
+            _key = new Lazy<SafeRsaHandle>(() => rsaHandle);
```

Comment:
+1 to the `public Lazy(T value)` ctor.

In the meantime, there is a [trick](https://gist.github.com/AArnott/d285feef75c18f6ecd2b) that can be used to avoid the closure allocation.

With the following helper:

```c#
internal static class FuncHelper
{
    internal static Func<T> AsFunc<T>(this T value) where T : class
    {
        return new Func<T>(value.Return);
    }

    private static T Return<T>(this T value)
    {
        return value;
    }
}
```

You can then do this:

```c#
_key = new Lazy<SafeRsaHandle>(rsaHandle.AsFunc());
```

Which reduces the allocations from delegate + closure, to just delegate.

Not necessarily suggesting that it should be used here, but figured I'd mention it.